sinks/signalfx/signalfx.go

```diff
@@ -183,11 +201,91 @@ func NewClient(endpoint, apiKey string, client *http.Client) DPClient {
 	return httpSink
 }
 
+func MigrateConfig(conf *veneur.Config) error {
+	if conf.SignalfxAPIKey.Value != "" {
+		return nil
+	}
+	if conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod == "" {
+		conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod = "10m"
+	}
+	dynamicPerTagAPIKeysRefreshPeriod, err :=
+		time.ParseDuration(conf.SignalfxDynamicPerTagAPIKeysRefreshPeriod)
+	if err != nil {
+		return err
+	}
+	conf.MetricSinks = append(conf.MetricSinks, struct {
+		Kind   string      "yaml:\"kind\""
+		Name   string      "yaml:\"name\""
+		Config interface{} "yaml:\"config\""
+	}{
+		Kind: "signalfx",
+		Name: "signalfx",
+		Config: SignalFxSinkConfig{
+			APIKey:                            conf.SignalfxAPIKey,
+			DynamicPerTagAPIKeysEnable:        conf.SignalfxDynamicPerTagAPIKeysEnable,
+			DynamicPerTagAPIKeysRefreshPeriod: dynamicPerTagAPIKeysRefreshPeriod,
+			EndpointAPI:                       conf.SignalfxEndpointAPI,
+			EndpointBase:                      conf.SignalfxEndpointBase,
+			FlushMaxPerBody:                   conf.SignalfxFlushMaxPerBody,
+			HostnameTag:                       conf.SignalfxHostnameTag,
+			MetricNamePrefixDrops:             conf.SignalfxMetricNamePrefixDrops,
+			MetricTagPrefixDrops:              conf.SignalfxMetricTagPrefixDrops,
+			PerTagAPIKeys:                     conf.SignalfxPerTagAPIKeys,
+			VaryKeyBy:                         conf.SignalfxVaryKeyBy,
+		},
+	})
+	return nil
+}
+
+func CreateSignalFxSink(
```

Comment:
I understand why the code is structured to make `CreateSignalFxSink` and `NewSignalFxSink` distinct (the former is copied from `server`) - but it's not super clear why they need to continue to be distinct now that we've moved this logic into the sink code.

Reply:
I think this might help clarify things a little, while we wait for future refactors:

- Unexport `newSignalFxSink` (to clarify that this is a package specific function, with nothing `server` cares about)
- Move as much of `Create` into `new` as we can